TwoNote/app/src/main/java/com/microsoft/device/display/wm_samples/twonote/MainActivity.kt

```diff
@@ -96,7 +95,19 @@ class MainActivity :
                 windowInfoRep.windowLayoutInfo
                     .collect { newLayoutInfo ->
                         dualScreenVM.isDualScreen = false
-                        val noteSelected = savedNote != null && savedINode != null
+                        var noteSelected = savedNote != null && savedINode != null
+
+                        // If no note/inode were passed through the saved instance, check if a
+                        // NoteDetailFragment exists and extract its note/inode
+                        if (!noteSelected) {
+                            try {
+                                val detailFragment =
+                                    supportFragmentManager.fragments.first { fragment -> fragment as? NoteDetailFragment != null }
+                                savedNote = detailFragment.arguments?.getSerializable(NOTE) as? Note
+                                savedINode = detailFragment.arguments?.getSerializable(INODE) as? INode
+                                noteSelected = true
+                            } catch (e: NoSuchElementException) {}
```

Comment:
If a good practice that if you catch the exception also add a Log so it shows what's going on whenever the issue happens.